### ISSUE TITLE
Alternate ranking system option

### ISSUE THREAD

**Issue by ephun** [spalmurray/wordle-bot#5]
Would it be possible to add a leaderboard that is a combination of games played and average score? So ranked in terms of "lowest average" and" highest played". On my family server, we have a few people that have only done it once or twice and it takes up space. I understand that you can delete users' data, this would be a cool option though!

**Comment by ephun** [comment]
Not sure how to flare sorry about that

**Comment by spalmurray** [comment]
Hey @ephun! I'm not sure I understand the leaderboard you're describing- could you give a concrete example of what data it would show and how it would order players?

**Comment by ephun** [comment]
Hey! Sorry, I apologize for my lack of specificity. 

On my discord server, 15 or so people regularly play and 4 or 5 play occasionally or have only played once or twice.

Some of the infrequent players scored really high—they only did one or two and have an average of 2.5 or something.

The infrequent players are also usually infrequent on the discord, so we can't really have leaderboard (the average guess number) made up of the lowest scores by frequent users.

So what I'm saying is it would be nice to have some sort of "culled leaderboard" functionality. I don't know what that would look like:
- Only appear on the leaderboard if you have solved at least 5 Wordles?
- Only appear on the leaderboard if you have solved the last 5 wordles?
- Some sort of mixture between "average" and "number of games played"?
- Score standard deviation?

I don't know how to quantify it. It would just be nice to have a leaderboard that reflects both quickness and consistency.

I think one of my friends might be attempting to make this in a fork, so I'll close it. Just something to think about, though!

**Comment by ephun** [comment]
Actually, I won't close it for now. But feel free to haha

**Comment by ephun** [comment]
Win rate could also be a factor? Not sure

**Comment by spalmurray** [comment]
Okay I understand what you're looking for now. This makes a lot of sense.

What about an optional parameter in the leaderboard commands that specifies how many scores a player must have to be included?

Something like `!wb average played=10` to include only those with 10 or more games played.

I've been thinking of adding an optional parameter to specify the number of entries to display in a leaderboard too (currently caps out at 10). I think an optional parameters system could be extended a lot to provide much more specific queries.

What do you think?

**Comment by ephun** [comment]
That sounds great!

**Comment by ephun** [comment]
Also, it'd be cool if `!wb me` told you your ranking

**Comment by spalmurray** [comment]
Yes I have considered that also. It would require significantly more resources, which was my main reservation, but I think it would be really nice. I will add some way to get this info. :)
